students/Josh_HOff/lesson09/mailroom_lesson_09.py

```diff
@@ -4,70 +4,149 @@
 
 donors = {'Josh Hoff': [25, 75], 'Tatsiana Kisel': [35, 105.55]}
 
+#This class handles all input from the user
+class Input(object):
+
+#    donors = {'Josh Hoff': [25, 75], 'Tatsiana Kisel': [35, 105.55]}
 
-class Donor(object):
     def __init__(self=None):
         pass
-        
+    
     def thank_you(self=None):
         while True:
             donor_name = input('\nWhat is the name of the donor?: ')
             if donor_name == 'list':
-                Donor().show_list()
+                DonorCollection().show_list()
                 continue
+            elif donor_name == 'quit':
+                break
             while True:
                 try:
                     donation = float(input('\nWhat is the donation amount?: '))
                 except ValueError:
                     print('\nPlease give a number instead.')
                     continue
                 break
-            Donor().add_donor(donor_name, donation)
+            DonorCollection().add_donor(donor_name, donation)
             break
+            
+    def quitting(self):
+        sys.exit()
+        
+    def continuing(self):
+        print('Try Again.\n')
+
+#this class handles all donor management
+class Donor(object):
+    def __init__(self=None, name = ''):
+        global donors
+        self._donors = donors
```

Comment:
this class is for a single donor object; what is the purpose of donors attribute?